### PR TITLE
Feature/ Support regex path names

### PR DESCRIPTION
**Pull request by buoyantair**
I was reading through this plugin yesterday and thought "Hey, why not support regex, that way I dont have to pass in an array of names!" and I tried to implement it. 

The tests don't work out at all, I am having a hard time debugging them, but I think I'm on the right path. 
This is not complete but If there''s anyone willing to help out and write these tests, it would be awesome! 

Thanks =3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -86,6 +107,16 @@ var clean = function() {
     webpackDir = upperCaseWindowsRoot(webpackDir);
   }
 
+  // Resolve RegExp paths.
+  _this.paths.reduce((acc, currentPath) => {
```
Comment (aulisius) [inline index.js]: You should be assigning the result of this `.reduce` back again to `_this,paths`. Better else, You can instead map the regex to resolved paths and assign it to `_this.paths`

File: index.js [hunk 2]
```diff
@@ -52,6 +52,27 @@ function CleanWebpackPlugin(paths, options) {
...
+ * @param {string} root 
+ * @param {RegExp} pattern 
+ * @returns [string]
+ */
+function resolveRegexPaths(root, pattern) {
+  const rootChildren = fs.readdir(root);
+  
+  const resolvedPaths = [];
+
+  rootChildren.forEach(child => {
```
Comment (aulisius) [inline index.js]: The codebase is very ES5 right now. I believe we can maintain one style for now. We can later on rewrite the entire plugin in one go with ES6.

File: index.js (hunk 1 above)
Comment (buoyantair) [reply index.js]: Hey, okay I'll look into it!

File: index.js (hunk 2 above)
Comment (buoyantair) [reply index.js]: Okay :D

**Comment by buoyantair** [comment]
@aulisius I resolved those issues ^-^, I'll work on the tests now.

**Comment by buoyantair** [comment]
@aulisius This feature is now working thanks to @Jarrku ! Please take a look :)

File: index.js [hunk 3]
```diff
@@ -54,23 +54,20 @@ function CleanWebpackPlugin(paths, options) {
...
 function resolveRegexPaths(root, pattern) {
-  const rootChildren = fs.readdir(root);
-  
-  const resolvedPaths = [];
+  var rootChildren = fs.readdirSync(root);
 
-  rootChildren.forEach(function (child) {
-    if (pattern.test(child) === true) {
-      resolvedPaths.push(child);
+  return rootChildren.reduce(function(paths, child) {
```
Comment (aulisius) [inline index.js]: This can be written with `.filter` as
```js
return rootChildren.filter(pattern.test)
```

**Comment by aulisius** [comment]
All looks good. Thoughts? @chrisblossom @johnagan

**Comment by DenizTemizer** [comment]
Some updates on this? Imo a must have feature for this plugin.

**Comment by buoyantair** [comment]
@DenizTemizer waiting for @chrisblossom and @johnagan to review I guess.

**Comment by aulisius** [comment]
Ping.

**Comment by aulisius** [comment]
Ping @chrisblossom @johnagan

**Comment by buoyantair** [comment]
I will just close it if no one comments :/

**Comment by johnagan** [comment]
Hey @buoyantair 👋 thank you for the PR. I'm just getting around to using webpack again and I'll check this PR out soon.

**Comment by chrisblossom** [comment]
Closed in [`v2.0.0`](https://github.com/johnagan/clean-webpack-plugin/tree/v2.0.0) via #99. See [johnagan/clean-webpack-plugin#usage](https://github.com/johnagan/clean-webpack-plugin#usage). Please open a new issue if you continue to have problems.

**Pull request closed without merging**